src/containers/app/EditAppType.js

```diff
@@ -0,0 +1,180 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Immutable from 'immutable';
+
+import { FormControl, FormGroup, ControlLabel, Button, Alert } from 'react-bootstrap';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import AsyncContent from '../../components/asynccontent/AsyncContent';
+import { editAppTypeRequest } from './AppActionFactory';
+
+
+class EditAppType extends React.Component {
+
+  static propTypes = {
+    actions: PropTypes.shape({
+      editAppTypeRequest: PropTypes.func.isRequired
+    }).isRequired,
+    editAppTypeAsyncState: PropTypes.instanceOf(Immutable.Map).isRequired,
+    id: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
+    namespace: PropTypes.string.isRequired,
+    title: PropTypes.string.isRequired,
+    description: PropTypes.string.isRequired,
+    entityTypeId: PropTypes.string.isRequired
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      description: '',
+      name: '',
+      namespace: '',
+      entityTypeId: ''
+    };
+  }
+
+  onTitleChange = (event) => {
```

Comment:
For all of these `onChange` helpers, I recommend doing `.trim()` on the value to avoid blank strings.